packages/discovery-provider/plugins/pedalboard/apps/trending-challenge-rewards/src/rewards.ts

```diff
@@ -0,0 +1,139 @@
+    specifier = startBlockSpecifier
+  } else {
+    const response = await getTrendingChallengesByDate(db, targetSpecifier)
+    const challenge = response[0]
+    completedBlock = challenge.completed_blocknumber! - 1
+    specifier = challenge.specifier
+  }
+  
+  // Fetch all undisbursed challenges
+  const endpoint = await sdk.services.discoveryNodeSelector.getSelectedEndpoint()
+  console.log('endpoint = ', endpoint)
+  const res = await axios.get(
+    `${endpoint}/v1/challenges/undisbursed?completed_blocknumber=${completedBlock}`
+  )
+  const data: Challenge[] = res.data.data
+  const toDisburse = data.filter((c) =>
+    ['tt', 'tp', 'tut'].includes(c.challenge_id)
+)
+
+// Claim all undisbursed challenges
+for (const challenge of toDisburse) {
+  const challengeId = Object.values(ChallengeId).find(id => id === challenge.challenge_id)!
+  console.log('Claimable challengeId = ', challenge)
+  let res
+  let retries = 10
+  while (retries > 0) {
+    try {
+      if (!dryRun) {
+        res = await sdk.challenges.claimReward({
+          challengeId,
+          userId: challenge.user_id,
+          specifier: challenge.specifier,
+          amount: parseFloat(challenge.amount),
+        })
+        console.log('res = ', res)
+        break // Success - exit retry loop
+      }
+    } catch (e) {
+      console.error(`Error claiming reward, challengeId = ${challengeId}, attempt ${11 - retries} of 10, error = `, e)
+      retries--
+      if (retries === 0) {
+        console.error(`Failed to claim reward after 10 attempts for challengeId = ${challengeId}`)
+      }
+    }
+  }
+}
+  
+  // Look at database to see if all challenges have been disbursed
+  const trimmedSpecifier = specifier.split(':')[0]
+  console.log('trimmedSpecifier = ', trimmedSpecifier)
+  const friendly = await getChallengesDisbursementsUserbanksFriendlyEnsureSlots(
+    db,
+    trimmedSpecifier
+  )
+
+  // Format the results for Slack
+  const formattedResults = formatDisbursementTable(friendly)
+  console.log(formattedResults)
+
+  if (slackChannel === undefined) return new Err('SLACK_CHANNEL not defined')
+  await client.chat.postMessage({
+    channel: slackChannel,
+    text: '```' + formattedResults + '```'
+  })
+
+  return new Ok(undefined)
+}
+
+export const findStartingBlock = async (
+  db: Knex
+): Promise<Result<[number, string], string>> => {
+  const challenges = await getTrendingChallenges(db)
+  const firstChallenge = challenges[0]
+  if (firstChallenge === undefined)
+    return new Err(`no challenges found ${challenges}`)
+  const completedBlocknumber = firstChallenge.completed_blocknumber
+  if (completedBlocknumber === null)
+    return new Err(`completed block number is null ${firstChallenge}`)
+  const specifier = firstChallenge.specifier
+  // why we subtract one
+  // https://www.notion.so/audiusproject/Manually-Complete-Rewards-Challenge-Manually-disburse-trending-24daed058dc54e4a8adb4912814481f2?pvs=4#1ced75c9c41740ce8facd19bf0d46720
```

Comment:
nit: would prefer we just put a shortened version of whatever text is on the other end of that link here.

Reply:
this shouldn't even be checked in, but also not my code 😓